snuba_sdk/expressions.py

```diff
@@ -130,44 +130,163 @@ class Function(Expression):
     def is_aggregate(self) -> bool:
         return is_aggregation_function(self.function)
 
-    def validate(self) -> None:
-        if not isinstance(self.function, str):
-            raise InvalidExpression(f"function '{self.function}' must be a string")
-        if self.function == "":
+    def accept(self, visitor: Visitor[TVisited]) -> TVisited:
+        return visitor.visit_function(self)
+
+
+class Op(Enum):
+    GT = ">"
+    LT = "<"
+    GTE = ">="
+    LTE = "<="
+    EQ = "="
+    NEQ = "!="
+    IN = "IN"
+    NOT_IN = "NOT IN"
+    LIKE = "LIKE"
+    NOT_LIKE = "NOT LIKE"
+
+
+@dataclass(frozen=True)
+class Condition(Expression):
+    lhs: Union[Column, Function]
+    op: Op
+    rhs: Union[Column, Function, ScalarType]
+
+    def accept(self, visitor: Visitor[TVisited]) -> TVisited:
+        return visitor.visit_condition(self)
+
+
+@dataclass(frozen=True)
+class Entity(Expression):
+    name: str
+    name_validator: re.Pattern[str] = field(
+        init=False, repr=False, compare=False, default=re.compile(r"[a-zA-Z_]+")
+    )
+
+    def accept(self, visitor: Visitor[TVisited]) -> TVisited:
+        return visitor.visit_entity(self)
+
+
+class InvalidExpression(Exception):
+    pass
+
+
+class InvalidEntity(Exception):
+    pass
+
+
+class Validation(Visitor[bool]):
```

Comment:
IF you are rising exceptions when the validation fails instead of returning false you should make this:
`class Validation(Visitor[None])`

otherwise you have three expected return conditions: True, False, exception, while you need only two.